operator/pkg/manifest/per_namespace_manifest_loader.go

```diff
@@ -39,17 +39,22 @@ func NewPerNamespaceManifestLoader(repo Repository) *PerNamespaceManifestLoader
 }
 
 func (p *PerNamespaceManifestLoader) ResolveManifest(ctx context.Context, o runtime.Object) (map[string]string, error) {
-	_, ok := o.(*corev1beta1.ConfigConnectorContext)
+	ccc, ok := o.(*corev1beta1.ConfigConnectorContext)
 	if !ok {
 		return nil, fmt.Errorf("expected the resource to be a ConfigConnectorContext, but it was not. Object: %v", o)
 	}
 
 	componentName := k8s.ConfigConnectorComponentName
 	channelName := k8s.StableChannel
-	v, err := ResolveVersion(ctx, p.repo, componentName, channelName)
-	if err != nil {
-		return nil, fmt.Errorf("error resolving the version for %v in %v channel: %w", componentName, channelName, err)
+
+	version := ccc.Spec.Version
```

Comment:
Don't we need to validate the version is "legal" here? How do we want to behave if its not? Set a status error and then call ResolveVersion() as normal?

Reply:
It will fail to load the manifest and stop reconciliation (and retry reconciliation with the normal backoff).

Reply:
The version is actually validated in the `ResolveVersion` below (more specific in k-d-p). Right, it will end up as a standard CR object status error telling users why it fails. Do you have any concerns about this? I feel like fall back to version-by-channel may not be what users want (otherwise they won't flip the version bit).